### DISCUSSION TITLE
Selling to demand without vehicles

### DISCUSSION THREAD

**Discussion by katepolak** [voxeltycoon/issues#475] [General]
# Suggestion
My idea on how this could be achieved would be using a separate building, a sort of "Market", that would try to sell all goods it has in it's storage to nearby demands, each market could only sell one type of goods, or maybe later the player could unlock bigger ones (supermarkets, hypermarkets) via research. Even the first market would be locked behind a research node though to balance it a bit.

**Comment by Stucimir** [comment]
I was thinking about mechanics close to how Labs work. You can place a warehouse and when there is business nearby which needs that resource, it can directly buy from there. And the stocking of the warehouse will be the same way as now by vehichles or belts. That way if the city expands a lot you can have a hub outside the city with warehouses for all the demands and transport them to warehouses distributed in the city by belts or vehicles.

**Comment by teenangst** [comment]
KISS
Inside a warehouse GUI permit it to sell goods to businesses.

It could be something that needs researching

**Comment by victorleblais** [comment]
To combine a bit of all the ideas here:

- Make a market that acts like a warehouse.
- It has a GUI to select which resources/business it will sell to.
- The building is unlocked by research (that should be around the mid-tier I guess)

**Comment by Garagoth** [comment]
There is already most of this in place in warehouses and logistics. It needs a bit of small additions:
- allow industries to take needed goods from warehouses in range using logistics (like from stations and how Lab works)
- add weights and priorities to logistics connections (for example 3/5 of goods to industry A, 2/5 to B) (I guess #179 covers that)

Warehouses can be already fed by vehicles via stations and directly by conveyor belts, optionally add an option for a conveyor belt connector to feed goods to stations (and goods will be stored in connected warehouse) (see also voxeltycoon/bug-tracker#203), but this depends on your vision where conveyor belts end and logistics begin (as those can overlap in functionality a bit).

**Comment by MajorFordson** [comment]
Allow me to +1 the idea of a mid-game building for delivering goods through a city, without the player having to set up a dozen trucks and truck stops. Being able to simply set up an area of effect building to sell goods that arrive on, say a mixed goods freight train into a train station, would be perfect for players who don't want to micro manage an entire fleet of trucks in every city. You can even slightly penalise such an idea by taking a percentage of the sales for the distributors.

For such a system it would be really nice to see delivery vans (not trucks) coming and going from such a distribution building out to each business, like delivery carriers in many city builder games.

**Comment by Tenybras** [comment]
I was completely confused that this was not the case, so for me it looked like a Bug and it halted my gameplay.

I see that a business is connected to a station and the station has the needed goods.
![image](https://user-images.githubusercontent.com/39140051/115861889-b3b1e680-a433-11eb-8379-624baea0b108.png)
![image](https://user-images.githubusercontent.com/39140051/115861921-bdd3e500-a433-11eb-8141-c7c411912973.png)
As you can see it _states_ it is outputting to the other building

To fix this I had park a car forever in front of the Station, which picked up 5 items from the Station and put them back in the station to be delivered.
Either the tutorial or tooltip should state that Goods can not be deliver to businesses without trucks
![image](https://user-images.githubusercontent.com/39140051/115862199-202ce580-a434-11eb-842c-c039683e64e3.png)

Request for a thread label to be added either one of these that would deemed fit by the developer: Bug/documentation/gameplay.

**Comment by greeny** [comment]
> As you can see it states it is outputting to the other building

@Tenybras yes, the **station** is outputting to a building, not the warehouse.

**Reply by prostolyubo** [reply]
But the station also states that it has X goods available in the warehouse it is connected to. So IMHO, it should output to the building without this "phantom" vehicle transporting.

**Reply by victorleblais** [reply]
Buildings can't "pass along" the items. Otherwise you could just build warehouses from the mine to the city and it would magically work.

**Reply by greeny** [reply]
> But the station also states that it has X goods available in the warehouse it is connected to. So IMHO, it should output to the building without this "phantom" vehicle transporting.

it says it has resources available for loading, not for magically transferring to other buildings.

**Comment by IDragonfire** [comment]
Missed this issue and created a new one with some different ideas: https://github.com/voxeltycoon/issues/discussions/394
Had a similar idea like @MajorFordson


### New:
* Add an upgrade to an company to
  * allow company to get goods from warehouse 
  * act as an space saving truck station

**Comment by prostolyubo** [comment]
Or simply add delivery vans, that go from the station to all the companies in range (that have the "output to this building" on). You would buy/sell them in the station GUI without any setup to not have to manually manage them nor create routes for them.
If a station has goods available in the connected warehouse, it should be able to sell them to the destination.

**Reply by victorleblais** [reply]
So, you mean trucks?... Like what we have in game?

**Reply by IDragonfire** [reply]
Vans and trucks are different vehicles...

**Comment by andrewpey** [comment]
We've addressed this issue in the `0.86.0.5` (available on the [beta branch](https://voxeltycoon.xyz/devlog/beta-branch/)) by changing some texts at the station window and adding a tip that clarifies stations' purpose to its Logistics tab.

While this solution should help players to understand how things are supposed to work, we'll keep an eye on the discussion, and see if we can improve stations' behavior any further.

**Comment by kalmix88-hun** [comment]
I already thought I should start a new warehouse entry, but I'll write here what I thought. The game lacks a type of warehouse that can hold multiple goods at once, or from which stores can take goods. This is how warehouses work in Industry Giant 2. I'm sure this exists in other games, but this is what came to mind right now.
